src/solver/trunk.jl

```diff
@@ -169,12 +174,22 @@ function trunk(nlp :: AbstractNLPModel;
     verbose && @printf("%4d  %9.2e  %7.1e  %7.1e  ", iter, f, ∇fNorm2, get_property(tr, :radius))
 
     optimal = ∇fNorm2 <= ϵ
-    tired = nlp.counters.neval_obj > max_f
+    elapsed_time = time() - start_time
+    tired = neval_obj(nlp) > max_f || elapsed_time > max_time
   end
   verbose && @printf("\n")
 
-  stalled || (status = tired ? "maximum number of evaluations" : "first-order stationary")
+  if !stalled && !tired
+    status = :first_order
+  elseif tired
+    if neval_obj(nlp) > max_f
+      status = :max_eval
+    elseif elapsed_time > max_time
+      status = :max_time
+    end
+  end
```

Comment:
I occasionally see situations where the problem is solved at the last iteration, so that `tired` and `optimal` are both true. We should be sure that the status is `:first_order` in such a situation.